File: typescript/src/renderer.ts
import { newRectangle } from "./shapes/rectangle.js";

function newRenderer(width: number, height: number) {
    return {
        draw() {
            const area: number = newRectangle(height, width).computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.Shape;

public class Renderer {
    private Shape shape;
    
    // Renderer(double height, double width) {
    //     this.rectangle = new Rectangle(height, width);
    // }

    // BUT WHAT IF you want to render other shapes? 
    // make Renderer a class using all the shapes! use the INTERFACE!!
    
    Renderer(Shape shape) {
        this.shape = shape;
    }

    void draw() {
        double area = shape.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
